modules/date/tests/test_date.c

```diff
@@ -183,6 +183,46 @@ Test(date, test_date_with_additional_text_at_the_end)
   log_msg_unref(logmsg);
 }
 
+Test(date, test_date_with_multiple_formats)
+{
+  static struct
+  {
+    const gchar *msg;
+    int usec;
+  }
+  params[] =
+  {
+    { "2017-02-02 00:29:16",                0 },
+    { "2017-02-02 00:29:16,706",       706000 },
+    { "2019-05-04T21:55:46.989+02:00", 989000 },
+  };
```

Comment:
Big thanks for the unit test!
Criterion has parametric test capabilities. A very simple example: https://github.com/balabit/syslog-ng/blob/master/tests/unit/test_findcrlf.c
Please instead of this loop either use that, or create the test one-by-one.